libs/langgraph/langgraph/pregel/remote.py

```diff
@@ -0,0 +1,458 @@
+from typing import (
+    Any,
+    AsyncIterator,
+    Iterator,
+    Optional,
+    Sequence,
+    Union,
+)
+
+import orjson
+from langchain_core.runnables import Runnable, RunnableConfig
+from langchain_core.runnables.graph import (
+    Edge as DrawableEdge,
+)
+from langchain_core.runnables.graph import (
+    Graph as DrawableGraph,
+)
+from langchain_core.runnables.graph import (
+    Node as DrawableNode,
+)
+from langchain_core.runnables.schema import StandardStreamEvent, StreamEvent
+from langgraph_sdk.client import LangGraphClient, SyncLangGraphClient
+from langgraph_sdk.schema import Checkpoint, ThreadState
+from typing_extensions import Self
+
+from langgraph.checkpoint.base import CheckpointMetadata
+from langgraph.pregel.protocol import PregelProtocol
+from langgraph.pregel.types import All, PregelTask, StateSnapshot, StreamMode
+from langgraph.types import Interrupt
+from langgraph.utils.config import merge_configs
+
+
+class RemotePregel(PregelProtocol, Runnable):
+    def __init__(
+        self,
+        client: LangGraphClient,
+        sync_client: SyncLangGraphClient,
+        graph_id: str,
+        config: Optional[RunnableConfig] = None,
+    ):
+        self.client = client
```

Comment:
is there some way we can accept a single client and build the other one from it?, or make only one of them required at least

Reply:
Most people will be using only sync or only async